tests/e2e/monitoring_test.go

```diff
@@ -1093,3 +1106,480 @@ func withMonitoringTraces(backend, secret, size, retention string) testf.Transfo
 
 	return testf.TransformPipeline(transforms...)
 }
+
+// ValidateNamespaceRestrictedMetricsAccess tests the namespace-restricted metrics access functionality.
+func (tc *MonitoringTestCtx) ValidateNamespaceRestrictedMetricsAccess(t *testing.T) {
```

Comment:
This suggests that also access is tested, but it is checking the correctness of resources deployed.